pkg/asset/cluster/azure/azure.go

```diff
@@ -38,7 +39,42 @@ func PreTerraform(ctx context.Context, clusterID string, installConfig *installc
 	if err != nil {
 		return errors.Wrap(err, "failed to get session")
 	}
-	return tagResourceGroup(ctx, clusterID, installConfig, session)
+
+	if err := tagResourceGroup(ctx, clusterID, installConfig, session); err != nil {
+		return err
+	}
+
+	if len(installConfig.Config.Azure.VirtualNetwork) == 0 {
+		return nil
+	}
+	return tagVNet(ctx, clusterID, installConfig, session)
+}
+
+func tagVNet(ctx context.Context, clusterID string, installConfig *installconfig.InstallConfig, session *icazure.Session) error {
+	vnetClient, err := armnetwork.NewVirtualNetworksClient(session.Credentials.SubscriptionID, session.TokenCreds, nil)
+	if err != nil {
+		return errors.Wrap(err, "failed to get the virtual network client")
```

Comment:
nit: we are trying to generally move the codebase away from `errors.Wrap` to `fmt.Errorf`,

Reply:
added a separated commit refactoring the entire file to use `fmt.Errorf` isntead of `errors.Wrap`